Tests/UnitTests/Mocks/MockSKProductDiscount.swift

```diff
@@ -7,17 +7,24 @@ import Foundation
 import StoreKit
 
 @available(iOS 11.2, macCatalyst 13.0, tvOS 11.2, macOS 10.13.2, *)
-class MockSKProductDiscount: SKProductDiscount {
+class MockSKProductDiscount: MockSKProductDiscountWithNoPriceLocale {
+
+    var mockLocale: Locale? = Locale(identifier: "USD")
+    override var priceLocale: Locale {
+        return self.mockLocale ?? super.priceLocale
+    }
+
+}
+
+// See https://github.com/RevenueCat/purchases-ios/issues/1521
+@available(iOS 11.2, macCatalyst 13.0, tvOS 11.2, macOS 10.13.2, *)
+class MockSKProductDiscountWithNoPriceLocale: SKProductDiscount {
```

Comment:
I had to make a separate `class` to be able to simulate this. Overriding the property was making it crash here without letting the extension detect the `nil` `priceLocale`.